explorer/client/src/components/view-more/ViewMore.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useState, useCallback } from 'react';
+
+import styles from './ViewMore.module.css';
+
+type Props = {
+    children: JSX.Element[];
+    limitTo?: number;
+    label?: string;
+};
+
+const DEFAULT_MIN_NUMBER_OF_ITEMS_TO_SHOW = 3;
+
+function ViewMore({ children, limitTo, label }: Props) {
+    const [viewMore, setVeiwMore] = useState(false);
+    const numberOfListItemsToShow =
+        limitTo || DEFAULT_MIN_NUMBER_OF_ITEMS_TO_SHOW;
+    const viewAll = useCallback(() => {
+        setVeiwMore(!viewMore);
+    }, [viewMore]);
+
+    const viewMoreLabel = label
+        ? `View ${children.length} ${label}`
```

Comment:
`children` is kind of special in React, so I'd generally avoid working with Children directly like this. There's a handful of `React.Children` to work with the object (e.g. https://reactjs.org/docs/react-api.html#reactchildrencount).

Although I think I'd just avoid using children altogether for this, and just make a different prop that accepts an array to avoid any potential confusion.

Reply:
Thanks for pointing that out. I will look into an alternative method.